pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -101,36 +99,17 @@ func getProjectAndZone() (string, string, error) {
 	return projectID, zone, nil
 }
 
-func getInstanceID() (string, error) {
-	result, err := metadata.Get("instance/hostname")
+func getNetworkName(svc *compute.Service, projectID string) (string, error) {
+	networkList, err := svc.Networks.List(projectID).Do()
```

Comment:
I'll fork the code to use Metadata to fetch network name when running inside the cluster, and a REST API call when outside (E2E).